server/index.js

```diff
@@ -169,18 +176,23 @@ module.exports = (_config = {}) => {
     );
     app.get('/server', (req, res) => res.status(200).send({ bcoinUri: uri }));
 
-    if (nodeClient) {
-      logger.info(`Connecting with ${clientConfig.str('client-id')} client`);
-      app.use('/bcoin', bcoinRouter(nodeClient));
-    }
+    app.use('/clients', clientRoutes(clients));
 
-    if (walletClient) {
-      app.use('/bwallet', bcoinRouter(walletClient));
-    }
+    // redirects to support old routes
+    app.use('/bcoin', (req, res) =>
+      res.redirect(307, `/clients/${clientConfig.str('id')}/node${req.path}`)
+    );
 
-    if (multisigWalletClient) {
-      app.use('/multisig', bcoinRouter(multisigWalletClient));
-    }
+    app.use('/bwallet', (req, res) =>
+      res.redirect(307, `/clients/${clientConfig.str('id')}/wallet${req.path}`)
+    );
+
+    app.use('/multisig', (req, res) =>
+      res.redirect(
+        307,
+        `/clients/${clientConfig.str('id')}/multisig${req.path}`
```

Comment:
This pattern seems like it could result in a security issue, it reminds me of sql injection attacks, if bpanel is required and the config is injected then an attacker could grab the request.

Maybe pull from the config once at runtime instead of pulling from it each time a request comes in as well

Reply:
Updated